### PR TITLE
CL-2 Added util file under core lib generator

### PR DESCRIPTION
**Pull request by oren-una**
Let me know if this is okay

### CODE REVIEW

File: core_lib_generator/template_core_lib/util/util.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import os
+import threading
+import traceback
```
Comment (shay-te) [inline core_lib_generator/template_core_lib/util/util.py:3]: please make sure also that the example test is using this file

**Review by shay-te — CHANGES_REQUESTED** [review]

File: core_lib_generator/core_lib_generator_from_yaml.py [hunk 2]
```diff
@@ -200,6 +200,10 @@ def generate_tests(self):
             {},
             TestConfigOverrideGenerateTemplate(),
         )
+        self._generate_template(f'{self.snake_core_lib_name}/tests/test_data/helpers/util.py',
```
Comment (shubham-shay-te) [inline core_lib_generator/core_lib_generator_from_yaml.py:203]: create a seperate generator for util, `TestUtilGenerateTemplate` this class will be responsible to generate the util and replace the template strings with the actual core lib name

File: core_lib_generator/config_collectors/data_access.py [hunk 3]
```diff
@@ -15,14 +15,14 @@ def is_exists_data_access(user_input: str):
     if db_entities:
         for entity in db_entities:
             name = entity['key']
-            conn = entity['db_connection']
```
Comment (shubham-shay-te) [inline core_lib_generator/config_collectors/data_access.py:18]: revert `git rm --cached path/to/file`

**Review by shubham-shay-te — CHANGES_REQUESTED** [review]

File: core_lib_generator/template_core_lib/tests/test_template.py [hunk 4]
```diff
@@ -2,11 +2,12 @@
...
 # template_test_imports
 from core_lib.helpers.test import load_core_lib_config
+from core_lib_generator.template_core_lib.tests.test_data.helpers.util import sync_create_start_core_lib
 
 
 class TestTemplate(unittest.TestCase):
     def setUpClass(self):
-        config = load_core_lib_config('./test_data/test_config/config.yaml')
-        self.template_snake_core_lib = TemplateCoreLib(config)
+        self.template_snake_core_lib = sync_create_start_core_lib()
```
Comment (shubham-shay-te) [inline core_lib_generator/template_core_lib/tests/test_template.py:10]: can we also update the `core_lib_generator/file_generators/test_generator.py` to remove any updates we are not making

File: core_lib_generator/file_generators/test_util_generator.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        core_lib_import: str = f'from {os.path.basename(os.getcwd())}.{core_lib_name}.{core_lib_name}.{core_lib_name} import {camel_case_name}'
+        template_content = template_content.replace('# template_core_lib_import', core_lib_import)
+
+        template_class: str = f'{camel_case_name}Instance'
+        template_content = template_content.replace('TemplateInstance', template_class)
+
+        return template_content
+
+    def get_template_file(self, yaml_data: dict) -> str:
+        return 'template_core_lib/tests/test_data/helpers/util.py'
```
Comment (shubham-shay-te) [inline core_lib_generator/file_generators/test_util_generator.py:22]: `'template_core_lib/tests/test_data/helpers/util.py'` can you please check if the file exists here and you have pushed I don't see the file under this path in the PR

File: core_lib_generator/template_core_lib/util/util.py
Comment (shubham-shay-te) [inline core_lib_generator/template_core_lib/util/util.py:1]: this file should be under `template_core_lib/tests/test_data/helpers/util.py`

File: core_lib_generator/file_generators/test_util_generator.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+from core_lib.helpers.string import snake_to_camel
+from core_lib_generator.file_generators.template_generator import TemplateGenerator
+
+
+class UtilTestGenerateTemplate(TemplateGenerator):
+    def generate(self, template_content: str, yaml_data: dict, core_lib_name: str, file_name: str) -> str:
+        camel_case_name: str = snake_to_camel(core_lib_name)
+
+        import_instance_string: str = f'from {core_lib_name}_instance import {camel_case_name}'
+        template_content = template_content.replace('# template core_lib_instance', import_instance_string)
```
Comment (shubham-shay-te) [inline core_lib_generator/file_generators/test_util_generator.py:11]: can you verify if the util is using this import or it is needed here, I don't see this string the util file in the template

**Review by shubham-shay-te — CHANGES_REQUESTED** [review]

File: core_lib_generator/template_core_lib/util/util.py
Comment (oren-una) [reply core_lib_generator/template_core_lib/util/util.py:1]: The file is there, just under core_lib_generator folder

**Pull request closed without merging**